src/common/android/deviceResolver.ts

```diff
@@ -0,0 +1,37 @@
+import {ChildProcess} from "../node/childProcess";
+import * as Q from "q";
+
+export interface IDevice {
+    id: string;
+    status: string;
+}
+
+export class DeviceResolver {
+
+    /**
+     * Gets the list of Android connected devices and emulators.
+     */
+    public getConnectedDevices(): Q.Promise<IDevice[]> {
+        let deferred = Q.defer<IDevice[]>();
+
+        let childProcess = new ChildProcess();
+        childProcess.execToString("adb devices")
+            .then(output => {
+                let devices = this.parseConnectedDevices(output);
+                deferred.resolve(devices);
+            });
+
+        return deferred.promise;
+    }
+
+    public parseConnectedDevices(input: string): IDevice[] {
```

Comment:
It might make sense to return a Dictionary instead of a list here, so you can more easily query for a specific device.